app.js

```diff
@@ -456,16 +457,30 @@ aws.initConfig(awsProps)
                         rescanCompilerSecs * 1000);
                 }
 
+                const sentrySlowRequestMs = ceProps("sentrySlowRequestMs", 0);
+
                 webServer
                     .set('trust proxy', true)
                     .set('view engine', 'pug')
                     .on('error', err => logger.error('Caught error in web handler; continuing:', err))
+                    // sentry request handler must be the first middleware on the app
+                    .use(Sentry.Handlers.requestHandler())
+                    // eslint-disable-next-line no-unused-vars
+                    .use(responseTime((req, res, time) => {
+                        if (sentrySlowRequestMs > 0 && time >= sentrySlowRequestMs) {
+                            Sentry.withScope(scope => {
+                                scope.setExtra('duration_ms', time);
+                                Sentry.captureMessage('SlowRequest', 'warning');
+                            });
+                        }
+                    }))
                     // Handle healthchecks at the root, as they're not expected from the outside world
                     .use('/healthcheck', new healthCheck.HealthCheckHandler(healthCheckFilePath).handle)
                     .use(httpRoot, router)
                     .use((req, res, next) => {
                         next({status: 404, message: `page "${req.path}" could not be found`});
                     })
+                    // sentry error handler must be the first error handling middleware
                     .use(Sentry.Handlers.errorHandler)
```

Comment:
Does the handler need to be in here twice? Will we get double entries?

Reply:
they are two separate handlers.

- `Sentry.Handlers.requestHandler` adds request information to the sentry context
- `Sentry.Handlers.errorHandler` catches exceptions and logs them to sentry

Reply:
D'oh! Didn't see they were different. Great!